src/components/CommentsForm.jsx

```diff
@@ -0,0 +1,103 @@
+import {  useState } from "react";
+import { useParams } from "react-router-dom";
+import { postComments } from "../utils/api";
+
+
+
+const CommentsForm =({commentsList, setCommentsList,})=>{
+const params =useParams();
+
+const articleId = params.article_id;
+
+const[userName, setUserName]=useState('tickle122')
+const[submitting, setSubmitting]= useState(false);
+const[body, setBody]=useState("")
+const [err, setErr] =useState(null);
+
+const[message, setMessage]= useState("");
+
+
+
+const handleSubmit =(event)=>{
+    event.preventDefault();
+  
+
+            const inputComment = {
+            
+            username: userName,
+                body: body,
+                votes:0,
+                    };
+              
+           setCommentsList((currentComments)=> {
+                    return ([inputComment, ...currentComments])})
+                    setMessage("We have updated your comment");
+                     setErr(null)
+                     setSubmitting(true);
+        postComments(articleId, inputComment).catch((err) => {
+        if(body === ""){
+        setMessage("Please enter a comment")}
+        else{
+            setCommentsList((currentComments) => { 
+          return  [...currentComments]}).then((copyArray)=>{
+          copyArray.shift();
+          })
+            setErr("Something went wrong!, please try again!");
+                                ;
+                      }
+              })
+              setBody("")
+              setSubmitting(false);
+                   
+               
+                };
+
+return (
+       <div className="commentsform" >  <form
+            
+            onSubmit={handleSubmit}> 
+            <h3 className="grid_item1" >Post a New Comment Here</h3>
+          
+              <input className="grid_item2"
+              type="text"
+              value={body}
+              maxLength="1000"
+              required
+              disabled={submitting}
+              onChange={(event)=> setBody(event.target.value)}
+              />
```

Comment:
Small thing, but could you change this to a textarea to enable multi line comments - just incase someone wants to post a bit more of a comment